src/unittest/python/timer_tests.py

```diff
@@ -0,0 +1,43 @@
+from unittest import TestCase, mock
+from vyu import timer
+
+
+class TimerTests(TestCase):
+
+    @mock.patch('vyu.timer.time.time')
+    def test_starts_timer(self, mock_time):
+        t = timer.Timer()
+        t.start()
+        self.assertEqual(t.t0, mock_time.return_value)
+        mock_time.assert_called_once_with()
+
+    def test_timer_is_not_running_when_intialized(self):
+        t = timer.Timer()
+        self.assertFalse(t.isrunning())
+
+    @mock.patch('vyu.timer.time.time')
+    def test_timer_is_running_when_started(self, mock_time):
+        t = timer.Timer()
+        t.start()
+        self.assertTrue(t.isrunning())
+
+    @mock.patch('vyu.timer.time.time')
+    def test_timer_is_not_running_after_clear(self, mock_time):
+        t = timer.Timer()
+        t.start()
+        t.clear()
+        self.assertFalse(t.isrunning())
+
+    @mock.patch('vyu.timer.time.time')
+    def test_time_when_time_is_up(self, mock_time):
+        mock_time.side_effect = [1, 5]
+        t = timer.Timer()
+        t.start()
+        self.assertTrue(t.check_time(2))
+
+    @mock.patch('vyu.timer.time.time')
+    def test_time_when_time_is_not_up_yet(self, mock_time):
+        mock_time.side_effect = [1, 2]
```

Comment:
A couple of questions:

1. I'm not sure if the `time.time()` being is returning a list `[1, 2]` or if the list object is now the mock value for `time.time()` 
2. I believe the the second position in the list is the one that is important for this function, so my guess right now would be that the first value in the list is the frame number?

Reply:
This is a strange design decision that they made in the mock module. If you want to do anything even slightly more complicated than just replacing a return value to always the same value, you have to do that through side effect. In this case, it is simply the following:
```python
>>> mock_time()
1
>>> mock_time()
2
````